dotCMS/src/main/java/com/dotcms/storage/FileSystemStorage.java

```diff
@@ -0,0 +1,265 @@
+package com.dotcms.storage;
+
+import com.dotcms.concurrent.DotConcurrentFactory;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.google.common.collect.ImmutableList;
+import com.liferay.util.FileUtil;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Future;
+
+/**
+ * Represents a Storage on the file system
+ * The groups here are folder previously defined, you can subscribe more by using {@link #addBucketMapping(String, File)}
+ * @author jsanca 
+ */
+public class FileSystemStorage implements Storage {
+
+    private final Map<String, File> buckets = new ConcurrentHashMap<>();
+
+    /**
+     * Adds a mapping between a bucket name and a file
+     * @param bucketName {@link String} bucket name
+     * @param file {@link File}
+     */
+    public void addBucketMapping (final String bucketName, final File file) {
+
+        this.buckets.put(bucketName, file);
+    }
+
+    @Override
+    public boolean existsGroup(final String groupName) {
+
+        return this.buckets.containsKey(groupName) && this.buckets.get(groupName).exists();
+    }
+
+    @Override
+    public boolean existsObject(final String groupName, final String objectPath) {
+
+        return this.existsGroup(groupName) && new File(this.buckets.get(groupName), objectPath).exists();
+    }
+
+    @Override
+    public boolean createGroup(final String groupName) {
+
+        throw new UnsupportedOperationException("On FileSystemStorage can not create buckets, they have to be previously defined");
+    }
+
+    @Override
+    public boolean createGroup(final String groupName, final Map<String, Object> extraOptions) {
+
+        throw new UnsupportedOperationException("On FileSystemStorage can not create buckets, they have to be previously defined");
+    }
+
+    @Override
+    public boolean deleteGroup(final String groupName) {
+
+        throw new UnsupportedOperationException("On FileSystemStorage can not delete buckets");
+    }
+
+    public boolean deleteObject(final String bucket, final String path) {
+
+        return new File(this.buckets.get(bucket), path).delete();
+    }
+
+    @Override
+    public List<Object> listGroups() {
+
+        return new ImmutableList.Builder<>().addAll(this.buckets.keySet()).build();
+    }
+
+    @Override
+    public Object pushFile(final String groupName,
+                       final String path,
+                       final File file,
+                       final Map<String, Object> extraMeta) {
+
+        if (!this.existsGroup(groupName)) {
+
+            throw new IllegalArgumentException("The bucketName: " + groupName +
+                    ", does not have any file mapped");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal ", does not have any file mapped" appears 4 times in this file; the first occurrence is on line 84](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)